pandas/core/dtypes/cast.py

```diff
@@ -1026,7 +1026,7 @@ def maybe_cast_to_datetime(value, dtype, errors="raise"):
             )
 
             if is_datetime64 and not is_dtype_equal(dtype, _NS_DTYPE):
-                if dtype.name in ("datetime64", "datetime64[ns]"):
+                if dtype <= np.dtype("M8[ns]"):
```

Comment:
What does `<=` do for numpy dtypes? Check if it's a subtype?

Reply:
Not clear to me either:

```
>>> "timedelta64[ns]" < np.dtype("m8[ns]")
False
>>> "timedelta64[us]" < np.dtype("m8[ns]")
True
>>> "timedelta64[ps]" > np.dtype("m8[ns]")
True
>>> "timedelta64" < np.dtype("m8[ns]")
True
```